### PR TITLE
fix: handle prompt response pairs in feature discovery

### PR DESCRIPTION
**Pull request by axiomofjoy**

### CODE REVIEW

File: src/phoenix/datasets/dataset.py [hunk 1]
```diff
@@ -328,6 +328,15 @@ def _parse_dataframe_and_schema(dataframe: DataFrame, schema: Schema) -> Tuple[D
             unseen_column_names,
         )
 
+    for schema_field_name in ["prompt_column_names", "response_column_names"]:
```
Comment (mikeldking) [inline src/phoenix/datasets/dataset.py:331]:
```suggestion
    for llm_schema_field_name in ["prompt_column_names", "response_column_names"]:
```

Comment (mikeldking) [inline src/phoenix/datasets/dataset.py:331]:
```suggestion
    for schema_field_name in LLM_SCHEMA_FIELD_NAMES:
```
Can be in a constants list

File: src/phoenix/datasets/dataset.py [hunk 2]
```diff
@@ -444,6 +453,23 @@ def _check_embedding_features_schema_field_for_excluded_columns(
...
+def _check_embedding_column_names_for_excluded_columns(
+    embedding_column_name_mapping: EmbeddingColumnNames,
+    column_name_to_include: Dict[str, bool],
+    unseen_column_names: Set[str],
+) -> None:
+    """
+    Check embedding features for excluded column names.
+    """
+    included_embedding_features = {}
+    included_embedding_features[""] = deepcopy(embedding_column_name_mapping)
```
Comment (mikeldking) [inline src/phoenix/datasets/dataset.py:465]: not sure I follow this syntax of setting it at the "" index. Why not one line on 464?

**Review by mikeldking — APPROVED** [review]

File: src/phoenix/datasets/dataset.py (hunk 2 above)
Comment (mikeldking) [reply src/phoenix/datasets/dataset.py:465]: also is the deep copy needed? Somehow things get accidentally preserved by reference?

Comment (axiomofjoy) [reply src/phoenix/datasets/dataset.py:465]: This was a mistake.

Comment (axiomofjoy) [reply src/phoenix/datasets/dataset.py]: Resolved in [87c4ed8](https://github.com/Arize-ai/phoenix/pull/609/commits/87c4ed8b0c2db8dcf09f42e72d566d3bbc56dc93)

File: src/phoenix/datasets/dataset.py (hunk 1 above)
Comment (axiomofjoy) [reply src/phoenix/datasets/dataset.py]: Resolved in [87c4ed8](https://github.com/Arize-ai/phoenix/pull/609/commits/87c4ed8b0c2db8dcf09f42e72d566d3bbc56dc93)

Comment (axiomofjoy) [reply src/phoenix/datasets/dataset.py]: Resolved in [87c4ed8](https://github.com/Arize-ai/phoenix/pull/609/commits/87c4ed8b0c2db8dcf09f42e72d566d3bbc56dc93)

**Pull request merged**